R/prophet.R

```diff
@@ -0,0 +1,1189 @@
+
+  return()
+}
+
+.prophetCreatePerformancePlotMse <- function(prophetPerformancePlots, options, prophetEvaluationResults) {
+  if (!is.null(prophetPerformancePlots[["prophetPerformancePlotMse"]])) return()
+  
+  prophetPerformancePlotMse <- createJaspPlot(title = "Performance Plot MSE", height = 320, width = 480)
+  prophetPerformancePlotMse$dependOn(c("performancePlotsMse"))
+  
+  prophetPerformancePlotMse$plotObject <- .prophetPerformancePlotFill(prophetEvaluationResults, options, type = "mse")
+  
+  prophetPerformancePlots[["prophetPerformancePlotMse"]] <- prophetPerformancePlotMse
+  
+  return()
+}
+
+.prophetCreatePerformancePlotRmse <- function(prophetPerformancePlots, options, prophetEvaluationResults) {
+  if (!is.null(prophetPerformancePlots[["prophetPerformancePlotRmse"]])) return()
+  
+  prophetPerformancePlotRmse <- createJaspPlot(title = "Performance Plot RMSE", height = 320, width = 480)
+  prophetPerformancePlotRmse$dependOn(c("performancePlotsRmse"))
+  
+  prophetPerformancePlotRmse$plotObject <- .prophetPerformancePlotFill(prophetEvaluationResults, options, type = "rmse")
+  
+  prophetPerformancePlots[["prophetPerformancePlotRmse"]] <- prophetPerformancePlotRmse
+  
+  return()
+}
+
+.prophetCreatePerformancePlotMape <- function(prophetPerformancePlots, options, prophetEvaluationResults) {
+  if (!is.null(prophetPerformancePlots[["prophetPerformancePlotMape"]])) return()
+  
+  prophetPerformancePlotMape <- createJaspPlot(title = "Performance Plot Mape", height = 320, width = 480)
+  prophetPerformancePlotMape$dependOn(c("performancePlotsMape"))
+  
+  prophetPerformancePlotMape$plotObject <- .prophetPerformancePlotFill(prophetEvaluationResults, options, type = "mape")
+  
+  prophetPerformancePlots[["prophetPerformancePlotMape"]] <- prophetPerformancePlotMape
+  
+  return()
+}
+
+.prophetPerformancePlotFill <- function(prophetEvaluationResults, options, type) {
+  
+  sampleDat <- prophet::performance_metrics(prophetEvaluationResults, metrics = type, rolling_window = -1)
+  meanDat   <- prophet::performance_metrics(prophetEvaluationResults, metrics = type, rolling_window = 0)
+  
+  if (options$crossValidationUnit == "weeks") {
+    sampleDat$horizon <- sampleDat$horizon / 7
+    meanDat$horizon   <- meanDat$horizon / 7
+  }
+
+  if (type == "mape") {
+    sampleDat$mape    <- sampleDat$mape*100
+    meanDat$mape      <- meanDat$mape*100
+  }
+
+  xBreaks <- pretty(sampleDat$horizon)
+  yBreaks <- pretty(sampleDat[[type]])
+  
+  p <- ggplot2::ggplot(data = sampleDat, mapping = ggplot2::aes_string(x = "horizon", y = type)) +
+  
+    ggplot2::geom_point(size = 3, color = "grey") +
+  
+    ggplot2::geom_line(data = meanDat, color = "darkred", size = 1.25) +
+
+    ggplot2::scale_x_continuous(name = gettext("Horizon"), breaks = xBreaks, limits = range(xBreaks)) +
+
+    ggplot2::scale_y_continuous(name = gettext(stringr::str_to_upper(type)), breaks = yBreaks, limits = range(yBreaks))
```

Comment:
any reason why not just use `toupper()`?

Reply:
I didn't know the function existed in base R...